bastionzero/service/agents/agents.go

```diff
@@ -0,0 +1,82 @@
+package agents
+
+import (
+	"context"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/service/targets"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/types"
+	"github.com/bastionzero/bastionzero-sdk-go/internal/client"
+	"net/http"
+)
+
+const (
+	agentsBasePath = "api/v2/agents"
+)
+
+// TargetsService handles communication with the agents endpoints of the
+// BastionZero API.
+//
+// BastionZero API docs:
+//
+// https://cloud.bastionzero.com/api/#tag--Agents
+
+type AgentsService client.Service
+
+type AgentDetails struct {
+	// Agent type.
+	AgentType AgentType `json:"type"`
+	// Last reported status of the agent.
+	AgentStatus AgentStatus `json:"status"`
+	// UTC timestamp indicating when the last agent status change event occurred.
+	LastStatusUpdate types.Timestamp `json:"lastStatusUpdate"`
+	// Version of the agent.
+	Version string `json:"version"`
+	// The region this agent is registered in.
+	Region string `json:"region"`
+	// Agent's base64-encoded public key.
+	PublicKey string `json:"publicKey"`
+	// Information related to the agent's current control channel.
+	ControlChannel targets.ControlChannelSummary `json:"controlChannel,omitempty"`
```

Comment:
Change this to `*targets.ControlChannelSummary` since `ControlChannel` can be `null`. Otherwise, when it's null the struct will be set to default values which is kind of confusing for when there is no control channel.

Also, you can remove the `omitempty` tag since it has no effect as this type is only unmarshaled.